examples/waspleau/Notes.md

```diff
@@ -0,0 +1,198 @@
+2021-12-16T02:18:54.224850+00:00 app[web.1]: npm ERR! Exit status 1
+2021-12-16T02:18:54.224925+00:00 app[web.1]: npm ERR!
+2021-12-16T02:18:54.224925+00:00 app[web.1]: npm ERR! Failed at the server@0.0.0 db-migrate-prod script.
+2021-12-16T02:18:54.224953+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
+2021-12-16T02:18:54.229064+00:00 app[web.1]:
+2021-12-16T02:18:54.229202+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
+2021-12-16T02:18:54.229274+00:00 app[web.1]: npm ERR!     /app/server/.npm/_logs/2021-12-16T02_18_54_225Z-debug.log
+2021-12-16T02:18:54.250005+00:00 app[web.1]: npm ERR! code ELIFECYCLE
+2021-12-16T02:18:54.250216+00:00 app[web.1]: npm ERR! errno 1
+2021-12-16T02:18:54.254548+00:00 app[web.1]: npm ERR! server@0.0.0 start-production: `npm run db-migrate-prod && NODE_ENV=production node ./src/server.js "npm" "run" "start-production"`
+2021-12-16T02:18:54.254618+00:00 app[web.1]: npm ERR! Exit status 1
+2021-12-16T02:18:54.254695+00:00 app[web.1]: npm ERR!
+2021-12-16T02:18:54.254741+00:00 app[web.1]: npm ERR! Failed at the server@0.0.0 start-production script.
+2021-12-16T02:18:54.254783+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
+2021-12-16T02:18:54.258329+00:00 app[web.1]:
+2021-12-16T02:18:54.258473+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
+2021-12-16T02:18:54.258545+00:00 app[web.1]: npm ERR!     /app/server/.npm/_logs/2021-12-16T02_18_54_255Z-debug.log
+2021-12-16T02:18:54.433588+00:00 heroku[web.1]: Process exited with status 1
+2021-12-16T02:18:54.568000+00:00 heroku[web.1]: State changed from starting to crashed
+2021-12-16T02:19:01.210559+00:00 app[api]: Deployed web (f5ce2d583e48) by user thelastemail@gmail.com
+```
+
+Had to delete and re-add database. Probably when I originally deployed with sqlite there it got into a funky state.
+
+## 7) App still not booting on Heroku
+
+```
+ 2021-12-16T02:34:30.070430+00:00 heroku[web.1]: Error R10 (Boot timeout) -> Web process failed to bind to $PORT within 60 seconds of launch
+2021-12-16T02:34:30.110821+00:00 heroku[web.1]: Stopping process with SIGKILL
+2021-12-16T02:34:30.279723+00:00 heroku[web.1]: Process exited with status 137
+2021-12-16T02:34:30.350174+00:00 heroku[web.1]: State changed from starting to crashed
+2021-12-16T02:34:31.160262+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=waspleau.herokuapp.com request_id=ed3e548f-56a7-4042-8465-1fc684002d7f fwd="75.60.240.117" dyno= connect= service= status=503 bytes= protocol=https
+```
+
+Is it because the `.env` file is being copied into ./wasp/build/server? Or Redis? Or startup function?
+
+Needed to specify REDIS_URL on Heroku: `const queue = new Queue('waspleau', process.env.REDIS_URL || 'redis://127.0.0.1:6379')`
+
+However, do we want to be copying over the `.env` at all? Prob ok.
```

Comment:
So right now, .env is supposed to be only for local development, and it is gitignored. You are supposed to manually ensure that ENV vars are correctly set on your hosting provider. That said, this is a space we might want to explore further, to possibly provide for a nicer experience.

Reply:
Yeah that makes sense. Although see above for my behavior that seems to show if you have a .env file in your project, when you run `wasp build` it does still get copied into .wasp/build/server/.env

Reply:
No, I think that is a bug, commented above!